ir/expression.def

```diff
@@ -298,6 +298,7 @@ class ArrayIndex : Operation_Binary {
     ArrayIndex {
         if (auto st = left ? left->type->to<IR::Type_Stack>() : nullptr)
             type = st->elementType; }
+    toString{ return left->toString() + "[" + right->toString() + "]"; }
```

Comment:
toString was defined in this way especially to highlight where the problem is in an expression involving an array indexing operation. If you print the whole index expression the user may not know what part of the expression is the cause. Before accepting this change we should check whether it impacts the usability of the error messages. Perhaps a better way to output this would be to use "[...]" to make it clear that the index expression is elided.

Reply:
Thank you for your comment, @mbudiu-vmw. 

I'm proposing this change with respect to the implementation of `IR::Member::toString()` [here](https://github.com/p4lang/p4c/blob/f40c5d92fd62ebe04c35003ad733005b0ef1535a/ir/expression.def#L281). `IR::Member` and `IR::ArrayIndex` are often nested in each other and in that case their different `toString()` implementations lead to weird results, for example: `Member([].$valid) / ArrayIndex([]) / Member(hdr.stack)`. The proposed change leads to, IMHO, a much better (and very useful) result: `Member(hdr.stack[0].$valid) / ArrayIndex(hdr.stack[0]) / Member(hdr.stack)`.

I've checked the test logs for relevant error messages and found the following differences:
```
-binary_e.p4(32): [--Werror=type-error] error: Array indexing [] applied to non-array type int<2>
+binary_e.p4(32): [--Werror=type-error] error: Array indexing a[2] applied to non-array type int<2>

-issue1696-1.p4(10): [--Wwarn=ordering] warning: []: the order of headers in deparser is not uniquely determined by parser!
+issue1696-1.p4(10): [--Wwarn=ordering] warning: hdr.stack[0]: the order of headers in deparser is not uniquely determined by parser!

-stack_e.p4(27): [--Werror=type-error] error: []: Negative array index -2
+stack_e.p4(27): [--Werror=type-error] error: stack[-2]: Negative array index -2
```
They seem OK to me. What do you think?